utopia-remix/app/models/project.server.ts

```diff
@@ -264,3 +264,27 @@ export async function updateGithubRepository(params: {
     },
   })
 }
+
+export async function getProjectExtraMetadataForEditor(params: {
+  projectId: string
+  userId: string
+}): Promise<ProjectMetadataForEditor | null> {
+  const project = await prisma.project.findUnique({
+    where: {
+      proj_id: params.projectId,
+      owner_id: params.userId,
+    },
+    select: {
+      ProjectAccessRequest: true,
+    },
+  })
+  if (project == null) {
+    return null
+  }
+
+  return {
+    hasPendingRequests: project.ProjectAccessRequest.some(
+      (r) => r.status === AccessRequestStatus.PENDING,
```

Comment:
Shouldn't this be part of the query to prisma?

Reply:
mmmh I think that if we put that in the query we cannot do further checks on the access request statuses down the line